pytensor/link/numba/dispatch/compile_ops.py

```diff
@@ -103,30 +104,48 @@ def deepcopy(x):
 @register_funcify_default_op_cache_key(IfElse)
 def numba_funcify_IfElse(op, **kwargs):
     n_outs = op.n_outs
+    as_view = op.as_view
 
-    if n_outs > 1:
+    if n_outs == 1:
 
         @numba_basic.numba_njit
-        def ifelse(cond, *args):
-            if cond:
-                res = args[:n_outs]
-            else:
-                res = args[n_outs:]
+        def ifelse(cond, x_true, x_false):
+            arr = x_true if cond else x_false
+            return arr if as_view else arr.copy()
```

Comment:
On second thought I guess we can get rid of the special case and stay with the codegen for every case now